gradle/wrapper/gradle-wrapper.properties

```diff
@@ -21,4 +21,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.4.1-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.4-bin.zip
```

Comment:
Out of curiosity, have you tried newer Gradle versions?
Newer shadow-plugin versions (which require at least Gradle 6), are noticeable faster - hence the question.

Reply:
I've tried version 7.1.1. I don't remember everything that failed exactly, but there were some removed API which needed migration and something else, maybe plugins what needed version update as well.

To avoid broadening scope of changes with tasks not related to build cache, I didn't change major version here.
Though this looks interesting to me and I might try migrating to version 7 later.

Reply:
fwiw I took a stab at upgrading to the latest Gradle version in https://github.com/apache/iceberg/pull/2826. We currently depend on another gradle plugin to be Gradle 7 compatible